### PR TITLE
REFACTOR-#5459: Install code linters through conda and unpin flake8

### LINKED ISSUE #5459: REFACTOR: install code linters through conda
Move forward towards environment consistency as suggested in https://github.com/modin-project/modin/issues/5208#issuecomment-1313603889.

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Anatoly Myachev <anatoly.myachev@intel.com>

## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5459 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/core/dataframe/algebra/binary.py [hunk 1]
```diff
@@ -86,7 +86,9 @@ def caller(
                     return query_compiler.__constructor__(
                         query_compiler._modin_frame.broadcast_apply(
                             axis,
-                            lambda l, r: func(l, r.squeeze(), *args, **kwargs),
+                            lambda left, right: func(
+                                left, right.squeeze(), *args, **kwargs
+                            ),
```
Comment (vnlitvinov) [inline modin/core/dataframe/algebra/binary.py:91]: why change?

File: modin/pandas/test/test_series.py [hunk 2]
```diff
@@ -2277,7 +2277,8 @@ def test_map(data, na_values):
 
     # Index into list objects
     df_equals(
-        modin_series_lists.map(lambda l: l[0]), pandas_series_lists.map(lambda l: l[0])
+        modin_series_lists.map(lambda list: list[0]),
+        pandas_series_lists.map(lambda list: list[0]),
```
Comment (vnlitvinov) [inline modin/pandas/test/test_series.py:2281]:
```suggestion
        modin_series_lists.map(lambda lst: lst[0]),
        pandas_series_lists.map(lambda lst: lst[0]),
```
better not shadow builtins; also why change?

**Review by vnlitvinov — COMMENTED** [review]
Why code changes?..

File: modin/pandas/test/test_series.py (hunk 2 above)
Comment (anmyachev) [reply modin/pandas/test/test_series.py:2281]: Forbidden variable name - `l`

**Review by vnlitvinov — APPROVED** [review]
LGTM!

**Review by YarShev — APPROVED** [review]

**Pull request merged**